src/ui/public/vislib/lib/tilemap_settings.js

```diff
@@ -0,0 +1,135 @@
+import uiModules from 'ui/modules';
+import _ from 'lodash';
+import marked from 'marked';
+import url from 'url';
+
+marked.setOptions({
+  gfm: true, // Github-flavored markdown
+  sanitize: true // Sanitize HTML tags
+});
+
+
+uiModules.get('kibana')
+  .service('tilemapSettings', function ($http, mapsConfig, $sanitize) {
+
+    const attributionFromConfig = $sanitize(marked(mapsConfig.deprecated.config.options.attribution));
+    const optionsFromConfig = _.assign({}, mapsConfig.deprecated.config.options, {attribution: attributionFromConfig});
+
+    /**
+     * Object to read out the map-service configuration
+     *
+     * @type {Object}
+     */
+    const mapSettings = {
+
+      _queryParams: {license: ""},//this curheprently a mandatory parameter. Remove when no longer the case.
+      _settingsInitialized: false,
+
+      //initialize settings with the default of the configuration
+      _url: mapsConfig.deprecated.config.url,
+      _options: optionsFromConfig,
+
+      /**
+       * Must be called before getUrl/getOptions can be called.
+       */
+      whenSettingsReady: async function () {
```

Comment:
I fixed the bug of having the settings-loaded twice. 

One thing I'm still on the fence about: do we really want to be able that the MapSettings get to be invalidated during the app's lifetime? It would introduce more wiring to handle these invalidations, which I don't think is necessary. In the current situation, everything needs to be configured during the addSetupWork bootstrap phase, imho this is sufficient and a reasonable restriction. I don't see a direct use-case of having this be configurable mid-way.